src/NLU.DevOps.ModelPerformance/NLUAccuracy.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace NLU.DevOps.ModelPerformance
+{
+    using ConsoleTables;
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+
+    /// <summary>
+    /// This class contains utils function for performance calculations
+    /// and printing
+    /// </summary>
+    public static class NLUAccuracy
+    {
+        /// <summary>
+        /// Calculates the precision from a confusion matrix
+        /// </summary>
+        /// <param name="cm">confusion matrix metrics</param>
+        /// <returns>The precision result</returns>
+        public static double Precision(this ConfusionMatrix cm)
+        {
+            return Divide(cm.TruePositive, cm.TruePositive + cm.FalsePositive);
+        }
+
+        /// <summary>
+        ///  Calculates the recall from a confusion matrix
+        /// </summary>
+        /// <param name="cm"> confusin matrix metrics</param>
+        /// <returns> The recall result</returns>
+        public static double Recall(this ConfusionMatrix cm)
+        {
+            return Divide(cm.TruePositive, cm.TruePositive + cm.FalseNegative);
+        }
+
+        /// <summary>
+        /// Calculates the f1 score from a confusion matrix
+        /// </summary>
+        /// <param name="cm"> confusion matrix metrics</param>
+        /// <returns> the f1 result</returns>
+        public static double F1(ConfusionMatrix cm)
+        {
+            var precision = cm.Precision();
+            var recall = cm.Recall();
+            var denominator = precision + recall;
+            return denominator != 0 ? 2 * (precision * recall) / denominator : 0;
+        }
+
+        /// <summary>
+        /// Calculates Precision, Recall and F1 Score
+        /// </summary>
+        /// <param name="cm">confusion matrix metrics</param>
+        /// <returns>List that contains the results calculated</returns>
+        public static List<double> CalcMetrics(ConfusionMatrix cm)
```

Comment:
Need to be public?

Reply:
no doesn't need to will change it.

Reply:
Will keep this one exposed at public and make the other ones private. 

---
In reply to: [377126475](https://github.com/microsoft/NLU.DevOps/pull/250#discussion_r377126475) [](ancestors = 377126475)